components/AirlineList/index.js

```diff
@@ -0,0 +1,76 @@
+export default function AirlineList({ sortedAirlines, unitSystem }) {
+  const convertUnits = (value, unitSystem) => {
+    return unitSystem === "metric" ? value : Math.round(value * 0.3937);
+  };
+
+  return (
+    <ul>
+      {sortedAirlines.map(
+        ({ id, name, personalItem, cabinBag, freeCabinBag }) => {
+          const {
+            length: personalItemLength,
+            width: personalItemWidth,
+            height: personalItemHeight,
+          } = personalItem;
+
+          const {
+            length: cabinBagLength,
+            width: cabinBagWidth,
+            height: cabinBagHeight,
+          } = cabinBag;
+
+          return (
+            <li key={id}>
+              <h2>{name}</h2>
+              <section>
+                <strong>Size Personal Item:</strong>
+                <p>
+                  {`${Math.round(
+                    convertUnits(personalItemLength, unitSystem)
+                  )}`}
+                  {" x "}
+                  {`${Math.round(
+                    convertUnits(personalItemWidth, unitSystem)
+                  )} `}
+                  {" x "}
+                  {`${Math.round(
+                    convertUnits(personalItemHeight, unitSystem)
+                  )} ${unitSystem === "metric" ? "cm" : "in"}`}
+                  {" | "}
+                  {Math.round(
+                    (personalItemLength *
+                      personalItemWidth *
+                      personalItemHeight) /
+                      1000
+                  )}{" "}
+                  l
+                </p>
+                <strong>Cabin Bag:</strong>
+                <p>
+                  {`${Math.round(convertUnits(cabinBagLength, unitSystem))}`}
+                  {" x "}
+                  {`${Math.round(convertUnits(cabinBagWidth, unitSystem))} `}
+                  {" x "}
+                  {`${Math.round(convertUnits(cabinBagHeight, unitSystem))} ${
+                    unitSystem === "metric" ? "cm" : "in"
+                  }`}
+                  {" | "}
+                  {Math.round(
+                    (cabinBagLength * cabinBagWidth * cabinBagHeight) / 1000
+                  )}{" "}
+                  l
```

Comment:
Calling functions within your jsx like this is at least wild and definitely super hard to follow.

In React, it's generally okay to call functions in JSX, but it's important to remember that these functions will be called on every render. If the function is computationally expensive or causes side effects, it could negatively impact the performance of your app or create unexpected behavior.

These functions are pretty lightweight so this should not be a problem, but it's  no method you should get used to.

Reply:
> Calling functions within your jsx like this is at least wild and definitely super hard to follow.
> 
> In React, it's generally okay to call functions in JSX, but it's important to remember that these functions will be called on every render. If the function is computationally expensive or causes side effects, it could negatively impact the performance of your app or create unexpected behavior.
> 
> These functions are pretty lightweight so this should not be a problem, but it's no method you should get used to.

Should I instead better make variables for the caluclations and import them from a dedicated place, similar to components? I would also like to simplify the structure of the code and I fully agree, the more I execute in the main components, the harder it is to tell from the code how the site is supposed to look.